### PR TITLE
Pipe: normalise things with eventSchedule

### PR DESCRIPTION
**Pull request by rhiaro**
I think this could do with some smaller unit tests to find edge cases for the more fiddly bits, but it has two full test events that pass so I think it basically works.

### CODE REVIEW

**Comment by rhiaro** [comment]
[Normalisations](https://trello.com/c/C70upiyi/10-normalisations)

File: src/lib/iso8601-duration/index.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
+/**
+ * Borrowed from https://github.com/tolu/ISO8601-duration/blob/master/src/index.js
```
Comment (odscjames) [inline src/lib/iso8601-duration/index.js:2]: Would be good to document exactly what version so that people upgrading later can look up changelogs etc

File: src/lib/pipes/normalise-schedule-pipe.js [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+                    //       It should also let input be yyyy-mm-dd and exclude any events that
+                    //       start in that 24 hour period.
+                    if(typeof schedule.exceptDate !== 'undefined'){
+                        const exceptDates = Utils.ensureArray(schedule.exceptDate);
+                        for(let exDate of exceptDates){
+                            rruleSet.exdate(new Date(exDate));
+                        }
+                    }
+
+                    // Only generate events two weeks from NOW (or until schedule ends, whichever is sooner)
```
Comment (odscjames) [inline src/lib/pipes/normalise-schedule-pipe.js:90]: That's interesting.

If an thing is published for 2 months starting 1st Aug on the 1st Aug, we get the data 2nd Aug and make events till 15th Aug, and then it's never updated again ..... won't the system never generate normalised events for the rest of Aug and Sept?

(If you think this should be issued and looked at later happy for this not to be a blocker)

Comment (rhiaro) [reply src/lib/pipes/normalise-schedule-pipe.js:90]: Maybe the right thing to do here is generate all of them, then filter it down to two weeks from now in the RPDE output at query time?

File: src/lib/pipes/normalise-schedule-pipe.js [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+  }
+
+  count(schedule){
+    return schedule.repeatCount;
+  }
+
+  eventsUntil(until){
+    // Generate events until two weeks from now, or until the end date,
+    // whichever is sooner.
+    let twoWeeks = new Date(Date.now() + 12096e5);
```
Comment (odscjames) [inline src/lib/pipes/normalise-schedule-pipe.js:226]: Can this be a new setting variable? I've tried to make every magic number be put in there for ease of code readability and changing later.

File: src/lib/pipes/normalise-schedule-pipe.js (hunk 2 above)
Comment (odscjames) [reply src/lib/pipes/normalise-schedule-pipe.js:90]: Is there any reason not to generate and RPDE them all straight up? Are we just worried about a bad bit of data declaring an event every hour for a year and filling the database or something?

**Comment by odscjames** [comment]
Just noting you have a commit "pipes: Tidy up comments " that changes more than just comments and I just wanted to check that was intentional. If yes, then approved for you

**Comment by rhiaro** [comment]
> Just noting you have a commit "pipes: Tidy up comments " that changes more than just comments and I just wanted to check that was intentional. If yes, then approved for you

yes, also fixes a misspelled variable name and a silly copypaste error. Didn't seem worth splitting out.

File: src/lib/pipes/normalise-schedule-pipe.js (hunk 2 above)
Comment (rhiaro) [reply src/lib/pipes/normalise-schedule-pipe.js:91]: I have spun this out into https://github.com/openactive/conformance-services/issues/93
